pyfar/classes/audio.py

```diff
@@ -614,9 +614,9 @@ def __init__(
             # check and set n_samples
             if n_samples is None:
                 warnings.warn(
-                    "Number of time samples not given, assuming an even "
-                    "number of samples from the number of frequency bins.")
-                n_samples = (data.shape[-1] - 1)*2
+                    f"Number of samples not given, assuming {n_samples} "
+                    f"samples from {data.shape[-1]} frequency bins.")
+                n_samples = max(1, (data.shape[-1] - 1)*2)
```

Comment:
Does it make sense to write {n_samples} in the f-string here? n_samples is None. I suggest to print soemthing like "Number of samples not given, assuming the number of samples from {data.shape[-1]} frequency bins."

Reply:
Good point, could be solved by changig the order of the warning and `n_samples=...`, right?

Reply:
Thanks for noting. Changed the order as suggested to be able to output the assumed number of samples